### PR TITLE
Moving assets from CDN's to our codebase in developer portal

### PR DESCRIPTION
**Pull request by damianpm**
**What this PR does / why we need it**:

Moving all external assets (from CDNs) to our codebase in Developer Portal

**Which issue(s) this PR fixes** 

https://issues.jboss.org/browse/THREESCALE-4041

**Verification steps** 

You may need to reset your database to test it locally:

`bundle exec rake db:reset MASTER_PASSWORD=p USER_PASSWORD=p`

### CODE REVIEW

**Comment by damianpm** [comment]
> What's with the images?

Same problem with the css, they were in the cdn, copied css files couldn't find them

**Comment by thomasmaas** [comment]
Ok, I think they were in cdn for performance reasons. Not sure if that will be a problem or if we use CDN mainly for assets uploaded by customers. Do you know @hallelujah ?

**Comment by damianpm** [comment]
> Ok, I think they were in CDN for performance reasons. Not sure if that will be a problem or if we use CDN mainly for assets uploaded by customers. Do you know @hallelujah ?

But I remember you said some on-prem clients may not have Internet access, is that true?

**Comment by thomasmaas** [comment]
Yes, this functionality was developed when we were saas only. So we need to know how it works

**Comment by hallelujah** [comment]
We can host them and point CDN in SaaS no worries.
Every assets hosted by us can be afterwards used in a CDN (well that's what a CDN is for ;))

File: lib/developer_portal/app/assets/stylesheets/plans_widget.css [hunk 1]
```diff
@@ -91,11 +95,11 @@
 }
 
 .plan-preview-box .features ul li.absent {
-	background: transparent url('/images/cross.png') no-repeat top left;
+	background: transparent url('cross.png') no-repeat top left;
```
[the remark this answers is not in the document]
Comment (damianpm) [reply lib/developer_portal/app/assets/stylesheets/plans_widget.css]: This images were located outside developer portal in `app/assets/images/`, now I copied them to `lib/developer_portal/app/assets/images`

[the remark this answers is not in the document]
Comment (damianpm) [reply lib/developer_portal/app/assets/stylesheets/plans_widget.css]: @hallelujah @thomasmaas  I deployed to preview if you want to test

[the remark this answers is not in the document]
Comment (hallelujah) [reply lib/developer_portal/app/assets/stylesheets/plans_widget.css]: Maybe we should not copy those images.

Let me help.

File: lib/developer_portal/app/assets/stylesheets/plans_widget.scss [hunk 2]
```diff
@@ -91,11 +91,11 @@
 }
 
 .plan-preview-box .features ul li.absent {
-	background: transparent url('/images/cross.png') no-repeat top left;
+	background: transparent image_url('cross.png') no-repeat top left;
```
Comment (damianpm) [inline lib/developer_portal/app/assets/stylesheets/plans_widget.scss]: @hallelujah digesting images with `image_url`

File: lib/developer_portal/app/views/developer_portal/main_layout.html.liquid [hunk 3]
```diff
@@ -14,9 +14,9 @@
 
     {% csrf %}
     {{ '/css/bootstrap.css' | stylesheet_link_tag }}
-    {{ '//netdna.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.css' | stylesheet_link_tag }}
+    {% cdn_asset /font-awesome/4.3.0/css/font-awesome.css %}
```
Comment (hallelujah) [inline lib/developer_portal/app/views/developer_portal/main_layout.html.liquid]: So we would need to document those for customers to know what assets we are hosting

Documentation should go to https://github.com/3scale/porta/blob/0b98c4d6620a6fea2f55d2fff64c674fb9764452/lib/developer_portal/lib/liquid/tags/cdn_asset.rb#L8


Provide in the description:

```
The assets you can use here are limited to:

- font-awesome/4.3.0/css/font-awesome.css
- font-awesome/4.3.0/css/font-awesome.min.css
- jquery-ui/1.11.4/jquery-ui.css 
- jquery-ui/1.11.4/jquery-ui.min.js
- jquery/1.7.1/jquery.min.js
etc ....
```

Comment (damianpm) [reply lib/developer_portal/app/views/developer_portal/main_layout.html.liquid]: Done [here](https://github.com/3scale/porta/pull/1468/files#diff-1030a186b26a17a9e79dd0d24e0178d5R19) @hallelujah

**Pull request merged**